inference-engine/src/inference_engine/cnn_network_ngraph_impl.cpp

```diff
@@ -331,7 +351,18 @@ CNNNetworkNGraphImpl::reshape(const std::map<std::string, std::vector<size_t>>&
         auto newParam = std::make_shared<::ngraph::op::Parameter>(param->get_element_type(), shape);
         newParam->set_friendly_name(param->get_friendly_name());
         _ngraph_function->replace_parameter(i, newParam);
+
+        // Update parameters in sub-functions
+        for (auto& sub_func : sub_functions) {
+            const auto sub_func_params = sub_func->get_parameters();
+            for (auto paramt_it = sub_func_params.begin(); paramt_it != sub_func_params.end(); ++paramt_it) {
+                if ((*paramt_it)->get_friendly_name() == newParam->get_friendly_name()) {
```

Comment:
I never thought that user can change the subgraph parameters (subgraphs exist in TensorIterator and Loop operations). Why do we need it? I am not sure that it is correct.

Reply:
^^as describe above it was solution to the secondary problem. Reverted.